js/core/component.js

```diff
@@ -1,19 +1,20 @@
-var Config = require("./config"),
-    extend = require("./utils/extend").extend,
-    Options = require("./options/index").Options,
-    convertRulesToOptions = require("./options/utils").convertRulesToOptions,
-    Class = require("./class"),
-    Action = require("./action"),
-    errors = require("./errors"),
-    commonUtils = require("./utils/common"),
-    typeUtils = require("./utils/type"),
-    Callbacks = require("./utils/callbacks"),
-    EventsStrategy = require("./events_strategy").EventsStrategy,
-    publicComponentUtils = require("./utils/public_component"),
-    PostponedOperations = require("./postponed_operations").PostponedOperations,
-
-    isFunction = typeUtils.isFunction,
-    noop = commonUtils.noop;
+import Config from './config';
+import { extend } from './utils/extend';
+import { Options } from './options/index';
+import { convertRulesToOptions } from './options/utils';
+import Class from './class';
+import Action from './action';
+import errors from './errors';
+import Callbacks from './utils/callbacks';
+import { EventsStrategy } from './events_strategy';
+import publicComponentUtils from './utils/public_component';
+import { PostponedOperations } from './postponed_operations';
+import { isFunction, isPlainObject, type, isDefined } from './utils/type';
+import { noop } from './utils/common';
+
+const getEventName = (actionName) => {
+    return actionName.charAt(2).toLowerCase() + actionName.substr(3);
```

Comment:
I'm not sure about it, but how about using some constants to make the the numbers more clear?
```suggestion
    return actionName.charAt(firstAfterPrefix ).toLowerCase() + actionName.substr(secondAfterPrefix);
    ...
const firstAfterPrefix = "dx".length;
const secondAfterPrefix = firstAfterPrefix + 1;
```